node/src/run_modes.rs

```diff
@@ -40,45 +42,125 @@ impl RunModes {
+            Enter => (),
+            LeaveRight => return 0,
+            LeaveWrong => return 1,
+        };
+
+        if let LeaveWrong = self.verify_privilege_level(privilege_required, &mode, streams) {
+            return 1;
         }
+
         match match mode {
             Mode::DumpConfig => self.runner.dump_config(args, streams),
-            Mode::Initialization => self.runner.initialization(args, streams),
-            Mode::Service => self.runner.run_service(args, streams),
+            Mode::Initialization => self.runner.run_daemon(args, streams),
+            Mode::Service => self.runner.run_node(args, streams),
         } {
-            Ok(exit_code) => exit_code,
-            Err(e) => {
-                short_writeln!(streams.stderr, "Configuration error");
-                e.param_errors.into_iter().for_each(|required| {
-                    short_writeln!(
-                        streams.stderr,
-                        "{} - {}",
-                        required.parameter,
-                        required.reason
-                    )
-                });
+            Ok(_) => 0,
+            Err(RunnerError::Numeric(e_num)) => e_num,
+            Err(RunnerError::Configurator(conf_e)) => {
+                Self::configurator_err_final_processing(conf_e, streams);
                 1
             }
         }
     }
 
-    fn args_contain_help_or_version(args: &[String]) -> bool {
-        args.contains(&"--help".to_string())
-            || args.contains(&"-h".to_string())
-            || args.contains(&"--version".to_string())
-            || args.contains(&"-V".to_string())
+    fn configurator_err_final_processing(error: ConfiguratorError, streams: &mut StdStreams) {
+        short_writeln!(streams.stderr, "Configuration error");
+        Self::write_unified_err_msgs(streams, error.param_errors)
+    }
+
+    fn ensure_help_or_version(
+        args: &[String],
+        mode: &Mode,
+        streams: &mut StdStreams<'_>,
+    ) -> EnterProgram {
+        match match match Self::is_help_or_version(args) {
+            false => return Enter,
+            true => mode,
+        } {
+            Mode::DumpConfig => app_config_dumper(),
+            Mode::Initialization => app_daemon(),
+            Mode::Service => app_node(),
+        }
+        .get_matches_from_safe(args)
+        {
+            Err(e) => Self::process_clap_error_that_may_contain_help_or_version(e, streams),
+            x => unreachable!("the sieve for 'help' or 'version' failed {:?}", x),
+        }
+    }
+
+    fn process_clap_error_that_may_contain_help_or_version(
+        clap_error: Error,
+        streams: &mut StdStreams<'_>,
+    ) -> EnterProgram {
+        match clap_error {
+            err if err.kind == clap::ErrorKind::HelpDisplayed
+                || err.kind == clap::ErrorKind::VersionDisplayed =>
+            {
+                short_writeln!(streams.stdout, "{}", err.message);
```

Comment:
Suggested comment:
// Note: clap version <put version number here> will never return an ErrorKind::VersionDisplayed error with the version in the .message field: instead it always writes the version to io::stdout and returns an empty String in the .message field. This looks like an error in clap, which we hope will be fixed eventually; if it ever is, this code will be here waiting.